From 5fd0fd77e293328f354e7f6ed22632ba6ff96593 Mon Sep 17 00:00:00 2001
From: Maria Matejka <mq@ucw.cz>
Date: Sun, 22 Dec 2024 22:26:44 +0100
Subject: [PATCH] CLI: Flushing tmp_linpool after every shown net.

There is no reason to keep the allocated objects through multiple nets.
---
 nest/rt-show.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/nest/rt-show.c b/nest/rt-show.c
index 3986da83d..aa9209ca5 100644
--- nest/rt-show.c
+++ nest/rt-show.c
@@ -282,8 +282,9 @@ rt_show_cont(struct cli *c)
     rt_show_table(d);
 
   RT_FEED_WALK(&d->tab->req, f)
-    if (f->count_routes)
-      rt_show_net(d, f);
+    TMP_SAVED
+      if (f->count_routes)
+	rt_show_net(d, f);
 
   if (rt_export_feed_active(&d->tab->req))
     rt_feeder_unsubscribe(&d->tab->req);
-- 
GitLab

